Retry #142

twang817 opened the pull request:
Re-creating #142 from a dedicated branch.

vladdoster reviewed (approved):
Thanks for working through the zsd issue @twang817!

vladdoster reviewed (changes requested):
- [ ] add unit test for `link` ice in `ices` tests
- [ ] update [zinit README.md](https://github.com/zdharma-continuum/zinit/blob/main/README.md) documenting `link` ice
- [ ] - [ ] update [zinit gh pages gallery](https://github.com/zdharma-continuum/zinit/tree/gh-pages) with example usage of `link` ice.

twang817 commented:
Added a quick fix for using realpath to create relative symlinks.  Also added a zunit test & a README update.

I'll need someone to kick off GH Actions to help me through the zsdocs stuff again.

twang817 commented:
I spent a lot of time trying to work with the docs.  Zshelldoc doesn't seem to want to generate docs in the same order as it does on the host.  It seems that in my environment, it sorts with the prefixes, whereas when running on the GH runners, it ignores the prefix for sorting.  This results in bigger diffs.

Secondly, I haven't had a lot of luck getting asciidoc's a2x to properly generate man pages.  I suspect it's a version issue, but the tools don't seem to be pinned.  I was able to get asciidoctor-pdf to generate pdfs from the adoc files.

twang817 commented:
Can anyone help here?  I'm able to generate the zsdocs, but there seems to be an inconsistency in sorting the docs and therefore a diff shows up.  I've tried emulating the github actions runner to see if I can match the versions of zsdoc & zsh, but haven't had any luck with it.

vladdoster commented:
Eeek!!

Sorry for such a late response. Could you pull in latest changes from main and try regenerating the docs? I have fixed it.

`make doc`

 @twang817

vladdoster commented:
I like your docker changes, too. Would you mind breaking those changes into a separate PR?

twang817 commented:
> I like your docker changes, too. Would you mind breaking those changes into a separate PR?

I'm not exactly sure which docker changes you're talking about.  I noticed that someone already dockerized much of the zsdoc process in zinit.
